.changeset/red-guests-melt.md

```diff
@@ -0,0 +1,5 @@
+---
+'@keystonejs/adapter-knex': patch
```

Comment:
```suggestion
'@keystonejs/adapter-knex': major
```

@timleslie happy with this if we change to major?

Reply:
Yep, definitely make it major, but I don't want to put this through until I've had a chance to review the changes that that this introduces and check whether we need to do anything differently.

Reply:
Also, we should include a link to the `pg` docs so that people can assess for themselves whether the upgrade will impact their system.

Reply:
**`pg@7.17.0` to `pg@8.0.3`**

Based on the [`pg` change log](https://github.com/brianc/node-postgres/blob/master/CHANGELOG.md#pg800) and a quick peruse of [the code](https://github.com/brianc/node-postgres/compare/pg@7.17.0...pg@8.0.3), it this jump shouldn't be especially disruptive. Likely worse case is needing to update a connection string or config option.

**`knex@0.20.6` to `knex@0.21.1`**

[Change log](https://github.com/knex/knex/blob/master/CHANGELOG.md). Looks very safe. The only two things that jumped out at me were...

* Knex returns native JS promises instead of Bluebird ones. This means that you no longer use such methods as map, spread and reduce on QueryBuilder instance.
* Breaking upstream change in pg-query-stream: `Changed stream.close to stream.destroy which is the official way to terminate a readable stream. This is a breaking change if you rely on the stream.close method on pg-query-stream...though should be just a find/replace type operation to upgrade as the semantics remain very similar (not exactly the same, since internals are rewritten, but more in line with how streams are "supposed" to behave).`

And of course you'd only hit them if you were using Keystones knex instance for some pretty specific stuff.